sql/planbuilder/dml_validate.go

```diff
@@ -94,26 +95,60 @@ func existsNonZeroValueCount(values sql.Node) bool {
 	return false
 }
 
+// validateColumns performs two checks. The first is insert and destination column
+// names. The insert column name must be valid, and we reject duplicate/conflicting
+// column names. The second check validates that we are not trying to modify a
+// read-only column (generated), which depends on first pairing source/destination
+// columns.
 func validateColumns(tableName string, columnNames []string, dstSchema sql.Schema, source sql.Node) error {
-	dstColNames := make(map[string]*sql.Column)
-	for _, dstCol := range dstSchema {
-		dstColNames[strings.ToLower(dstCol.Name)] = dstCol
-	}
-	usedNames := make(map[string]struct{})
-	for i, columnName := range columnNames {
-		dstCol, exists := dstColNames[columnName]
-		if !exists {
-			return sql.ErrUnknownColumn.New(columnName, tableName)
+	type namePos struct {
+		name string
+		pos  int
+		gen  bool
+	}
+
+	insCols := make([]namePos, len(columnNames))
```

Comment:
If we're targeting perf, I would imagine you want to put these into a pool to avoid the mallocs

Reply:
do they escape? would have thought this is stack allocated, with the simple structs and scoping